src/Chainweb/Miner/Miners.hs

```diff
@@ -87,41 +96,38 @@ localTest tmv gen miners bh =
         Just (BlockRate (Seconds n)) -> int n
         Nothing -> error $ "No BlockRate available for given ChainwebVersion: " <> show v
 
+    work :: BlockHeader -> IO BlockHeader
+    work bh = MWC.geometric1 t gen >>= threadDelay >> pure bh
+
 -- | A single-threaded in-process Proof-of-Work mining loop.
 --
-localPOW :: TMVar BlockHeader -> ChainwebVersion -> BlockHeader -> IO ()
-localPOW tmv v bh = do
-    HeaderBytes newBytes <- usePowHash v (\p -> mine p (_blockNonce bh) tbytes) hbytes
-    new <- runGet decodeBlockHeaderWithoutHash newBytes
-    atomically $ putTMVar tmv new
+localPOW :: LogFunction -> ChainwebVersion -> Miner -> CutDb cas -> IO ()
+localPOW lf v m cdb = runForever lf "Chainweb.Miner.Miners.localPOW" loop
```

Comment:
This could possibly be merged with `localTest`. They share some logic, but they're both so short by this point.